### PR TITLE
Updating SearchableTrees for v10

### PR DESCRIPTION
**Pull request by sofietoft**
Updated the C# code sample starting on line 73.

### CODE REVIEW

File: Extending/Section-Trees/Searchable-Trees/index-v10.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        /// </param>
+        /// <returns></returns>
+        IEnumerable<SearchResultEntity> Search(string query, int pageSize, long pageIndex, out long totalFound, string searchFrom = null);
+    }
+}
+```
+
+Your implementation needs to return an IEnumerable of `SearchResultEntity` items:
+
+```csharp
```
Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:53]:
```
using Umbraco.Cms.Core.Models.ContentEditing;
using System.Runtime.Serialization;

namespace My.Website
{
    public class SearchResultEntity : EntityBasic
    {
        public SearchResultEntity() { }
        /// <summary>
        /// The score of the search result
        /// </summary>
        [DataMember(Name = "score")]
        public float Score { get; set; }
    }
}
```

SearchResultEntity() throws an error "Must declare a body.."

File: Extending/Section-Trees/Searchable-Trees/index-v10.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    [DataMember(Name = "score")]
+    public float Score { get; set; }
+}
+```
+
+A `SearchResultEntity` consists of a Score (a Float value) identifying its relevance to the search term, and the set of `EntityBasic` properties that all Umbraco objects share: eg Name, Id, Udi, Icon, Trashed, Key, ParentId, Path, Alias, AdditionalData.
+
+#### Example implementation of ISearchableTree
+
+If we have a custom section Tree with alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```
Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:70]:
```suggestion
If we have a custom section Tree with the alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```
```suggestion
If we have a custom section Tree with alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```

Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:70]:
```suggestion
If we have a custom section Tree with the alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```
```suggestion
If we have a custom section Tree with alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```

Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:70]:
```suggestion
If we have a custom section Tree with the alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```
```suggestion
If we have a custom section Tree with alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```

**Review by eshanrnh — COMMENTED** [review]
Looks good 💪 The code works as expected 🙂 Just added a couple of suggestions to the code and grammar.

File: Extending/Section-Trees/Searchable-Trees/index-v10.md (hunk 2 above)
Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:70]:
```suggestion
If we have a custom section Tree with the alias 'favouriteThingsAlias' (see the [custom tree example](../trees.md)) then we could implement searchability by creating the following C# class in our site:
```

File: Extending/Section-Trees/Searchable-Trees/index-v10.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        }
+    }
+}
+```
+
+That's all we need, after an application pool recycle, if we now search in the backoffice we'll see matches from our custom 'Favourite Things' tree:
+
+![Content Section Dashboards](images/favouritethings-search-v8.png)
+
+Umbraco automatically finds any implementation of `ISearchableTree` in your site, and automatically configures it to be used for the custom section mentioned in the TreeAlias property. Be careful not to accidentally have two `ISearchableTree` implementations trying to search the 'same' TreeAlias, it's *one* `ISearchableTree` per TreeAlias.
```
Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:131]:
```suggestion
Umbraco automatically finds any implementation of `ISearchableTree` in your site and automatically configures it to be used for the custom section mentioned in the TreeAlias property. Be careful not to accidentally have two `ISearchableTree` implementations trying to search the 'same' TreeAlias, it's *one* `ISearchableTree` per TreeAlias.
```

File: Extending/Section-Trees/Searchable-Trees/index-v10.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+## Replacing an existing Section Tree Search
+
+Perhaps you want to change the logic for searching an existing section of the site, (why? - well you might have a 'company name' property on a MemberType in the Member section, and you want searches for that company name to filter the members who work there, the default implementation will only search on Member Name).
+
+Or perhaps you want to replace Examine search in the backoffice with an external Search Service, e.g. Azure Search. In a cloud hosted implementation you don't need to build the Examine indexes on each new server as your cloud hosting scales out.
```
Comment (eshanrnh) [inline Extending/Section-Trees/Searchable-Trees/index-v10.md:137]:
```suggestion
Or perhaps you want to replace Examine search in the backoffice with an external Search Service, e.g. Azure Search. In a cloud-hosted implementation you don't need to build the Examine indexes on each new server as your cloud hosting scales out.
```

File: Extending/Section-Trees/Searchable-Trees/index-v10.md (hunk 1 above)
Comment (sofietoft) [reply Extending/Section-Trees/Searchable-Trees/index-v10.md:53]: I am not entirely sure I understand what you mean here? 🤔

**Review by sofietoft — COMMENTED** [review]
Thanks for the review 🎉

File: Extending/Section-Trees/Searchable-Trees/index-v10.md (hunk 1 above)
Comment (sofietoft) [reply Extending/Section-Trees/Searchable-Trees/index-v10.md:53]: Ah, OK.

I'll run it by the community team to double-check it 💪

**Comment by sofietoft** [comment]
Had the samples verfied by the community team.

Merging it in.

**Pull request merged**